### ISSUE TITLE
[input-date-picker] - Component does not initialize with `value`.

### ISSUE THREAD

**Issue by richiecarmichael** [Esri/calcite-design-system#8157]
### Check existing issues

- [x] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates

### Actual Behavior

Component does not initialize ISO date. Only place holder text displayed.
```ts
          <calcite-input-date-picker
            value={this._departureDate} // "2023-01-01"
          />
```
Workaround:
```ts
          <calcite-input-date-picker
            valueAsDate={toLocalDate(this._departureDate)}
          />
```
where:
```ts
function toLocalDate(utcDate: string): Date {
  const date = new Date(utcDate);
  return new Date(date.getUTCFullYear(), date.getUTCMonth(), date.getUTCDate());
}
```

### Expected Behavior

Expe

### Reproduction Sample

None

### Reproduction Steps

None

### Reproduction Version

1.10.0

### Priority impact

p3 - want for upcoming milestone

### Calcite package

- [X] @esri/calcite-components
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

ArcGIS Maps SDK for JavaScript

**Comment by maxxxxxdlp** [comment]
@richiecarmichael is this a duplicate of https://github.com/Esri/calcite-design-system/issues/7014?

**Comment by richiecarmichael** [comment]
#7014 is specific to `calcite-input-number` whereas I'm seeing an issue with `calcite-input-date-picker`. As @JonUihlein mentioned [here](https://github.com/Esri/calcite-design-system/issues/7014#issuecomment-1731439755) the issues may be related. Let's keep this for now. I can assist with verifying/testing when #7014 is fixed.

**Comment by geospatialem** [comment]
Estimate updated to reflect additional team expertise, which should include updating the tests to allow sending in a full UTC string.

**Comment by geospatialem** [comment]
Blocked by the effort in PR https://github.com/Esri/calcite-design-system/pull/8402 to determine if the above is mitigate, or next steps to achieve a solution.

**Comment by jcfranco** [comment]
@richiecarmichael Would you mind testing this again? It looks like it was [fixed in 3.x](https://codepen.io/jcfranco/pen/gbOPZwX?editors=1000) (thanks for the simplified repro case from #7014, @maxpatiiuk!).

**Comment by richiecarmichael** [comment]
> [@richiecarmichael](https://github.com/richiecarmichael) Would you mind testing this again? It looks like it was [fixed in 3.x](https://codepen.io/jcfranco/pen/gbOPZwX?editors=1000) (thanks for the simplified repro case from [#7014](https://github.com/Esri/calcite-design-system/issues/7014), [@maxpatiiuk](https://github.com/maxpatiiuk)!).

Yes, just tested with calcite 3.0.2 and it looks good. Thanks! Closing with **bonus points**.
